static/script.js

```diff
@@ -223,3 +224,27 @@ function downloadURL(url, name) {
     document.body.removeChild(link);
     delete link;
 }
+
+function fileNamesAreValid(){
+    let invalidNames = []
+
+    for (let child of getFiles(document.getElementById('file-container'))) {
```

Comment:
Getting the file contents here is IMO unnecessary. I would move the `fileNamesAreValid` check below the `let filesContent = getFilesContent();` above and just pass the `filesContent` as a parameter to the function.

Reply:
Nevermind, the content doesn't contain the filename.